scraper/mapharma/mapharma.py

```diff
@@ -20,90 +24,59 @@
 DEFAULT_CLIENT = httpx.Client(timeout=timeout, headers=MAPHARMA_HEADERS)
 logger = logging.getLogger('scraper')
 
-insee = {}
-campagnes = {}
-
-
-def get_name(soup):
-    name = soup.find(class_='pharma-block').find('h3').text.strip()
-    return name
-
-
-def get_address(soup):
-    address = soup.find('div', {'class': 'mb-1 text-muted'}).text.strip()
-    return address
-
-
-def get_reasons(soup):
-    reasons = []
-    divs = soup.find_all('div', {'class': 'js-campagne-data'})
-    for div in divs:
-        if 'data-campagne-id' not in div.attrs or div.attrs['data-campagne-id'] == '0':
-            continue
-        campagneId = div.attrs['data-campagne-id']
-        options = div.find_all('option')
-        for option in options:
-            if 'value' not in option.attrs or option.attrs['value'] == '0':
+campagnes_valides = []
+campagnes_inconnues = []
+opendata = []
+
+
+def campagne_to_centre(pharmacy: dict, campagne: dict) -> dict() :
+    insee = departementUtils.cp_to_insee(pharmacy.get('code_postal'))
+    departement = departementUtils.to_departement_number(insee)
+    centre = dict()
+    centre['nom'] = pharmacy.get('nom')
+    centre['type'] = DRUG_STORE
+    centre['long_coor1'] = pharmacy.get('longitude')
+    centre['lat_coor1'] = pharmacy.get('latitude')
+    centre['com_nom'] = pharmacy.get('ville')
+    adr_voie = pharmacy.get('adresse')
+    adr_cp = pharmacy.get('code_postal')
+    adr_nom = pharmacy.get('ville')
+    centre['address'] = f'{adr_voie}, {adr_cp} {adr_nom}'
+    business_hours = dict()
+    horaires = pharmacy.get('horaires')
+    days = ['lundi', 'mardi', 'mercredi',
+            'jeudi', 'vendredi', 'samedi', 'dimanche']
+    for day in days:
+        for line in horaires.splitlines():
+            if day not in line:
                 continue
-            optionId = option.attrs['value']
-            optionName = option.text.strip()
-            reasons.append({'campagneId': campagneId, 'optionId': optionId, 'optionName': optionName})
-    return reasons
-
-
-def get_profile(url: str, client: httpx.Client = DEFAULT_CLIENT):
-    profile = {}
-    profile['reasons'] = []
+            business_hours[day] = line.replace(f'{day}: ', '')
+    centre['business_hours'] = business_hours
+    centre['phone_number'] = pharmacy.get('telephone')
+    centre['rdv_site_web'] = campagne.get('url')
+    centre['com_insee'] = departementUtils.cp_to_insee(pharmacy.get('code_postal'))
```

Comment:
Ces fonctions (cp_to_insee, encode etc.) vont trigger une exception si la valeur n'existe pas. get return None par defaut mais tu peux changer cette valeur.

Reply:
le problème c'est que si je n'ai rien dans code_postal ça devient compliqué de trouver une valeur de remplacement

Reply:
ouaip a ce moment la pourquoi pas juste lever une exception.

(Et uniquement des cas se presente alors il faudra, dans une PR ulterieur, regarder avec juste l'adresse/ville si on ne peut pas recuperer le CP.)

Reply:
if not pharmacy.get('code_postal'):
        raise ValueError('Absence de code postal')